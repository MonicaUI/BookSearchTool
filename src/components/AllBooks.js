import React, { useState } from 'react'
import './AllBooks.scss'
import Modal from './Modal'

function AllBooks({
    title,
    author,
    firstPublished,
    publisher,
    isbn,
    cover
}) {
    const [modalActive, setModalActive] = useState(false)

    return (
        <div>
            <div className="snippet" onClick={() => setModalActive(true)}>
                <div className="snippet__cover">
                    <img
                        className="snippet__image"
                        src={cover}
                        alt="no cover available"
                    />
                    <div className="snippet__coverImage"></div>
                </div>
                <div className="allBook__title">
                    <h3 className="snippet__book-title">{title}</h3>
                    <p className="snippet__book-author">{author}</p>
                    <p className="snippet__book-date">First published in {firstPublished}</p>
                </div>

            </div>
            <Modal
                active={modalActive}
                setActive={setModalActive}
                title={title}
                author={author}
                firstPublished={firstPublished}
                publisher={publisher}
                isbn={isbn}
                cover={cover}
            />
        </div>
    )
}

export default AllBooks