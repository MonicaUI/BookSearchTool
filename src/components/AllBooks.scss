.snippet {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    max-width: 50vw;
    margin: auto;
    margin-bottom: 0.75rem;
    padding-left: 1.25rem;
    padding-right: 0.75rem;
    background-color: white;
    border-radius: 0.625rem;
    -webkit-box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.2);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.2);
    height: 180px;
}

.snippet__title {
    -webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    padding-top: 1.25rem;
    padding-bottom: 1.25rem;
    width: 250px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100px;
    flex-wrap: wrap;
}

.allBook__title {
    display: block;
    width: 200px;
    padding-top: 50px;
}

.snippet__title>p {
    width: 125px;
    margin-left: 10px;
}

.snippet__cover {
    margin-top: 0.375rem;
    margin-bottom: 0.375rem;
    -webkit-box-flex: 0.4;
    -ms-flex-positive: 0.4;
    flex-grow: 0.4;
    border-radius: 5px;

}

.snippet .snippet__image {
    border-radius: 5px;
    display: block;
    margin: auto;
    width: 125px;
    background: url("../avatar_book.png");
    background-repeat: no-repeat;
    background-size: cover;
    height: 90%;
    margin-top: 10px;

}

.snippet .snippet__book-title {
    color: var(--color-text-title);
    font-size: 1.125rem;
    margin-bottom: 0.25rem;
}

.snippet .snippet__book-author {
    color: var(--color-text-main);
}

.snippet__book-date {
    padding-top: 10px;
    font-size: .75em;
    color: #666;
    font-family: "Lucida Grande", Verdana, Geneva, Helvetica, sans-serif;
}

.snippet:hover {
    -webkit-box-shadow: none;
    box-shadow: none;
}

.load-more-button {
    border: none;
    border-radius: 0.625rem;
    background-color: var(--color-dark);
    color: var(--color-text-button);
    display: none;
    margin: auto;
    margin-bottom: 1rem;
    padding: 0.625rem;
    font-size: 1rem;
    -webkit-box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.2);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.2);
}

.trending {
    display: flex;
    overflow: hidden;
}

.load-more-button:hover {
    -webkit-box-shadow: none;
    box-shadow: none;
}

.load-more-button:active {
    opacity: 0.9;
}

@media (min-width: 900px) {
    #Snippet {
        width: 60vw;
    }
}

$animationSpeed: 20s;


@keyframes scroll {
    0% {
        transform: translateX(0);
    }

    100% {
        transform: translateX(calc(-250px * 7))
    }
}

.trending__image {
    height: 200px;
    width: 150px;
    border-radius: 10px;
}

.slider {
    height: 200px;
    margin: 20px;
    position: relative;
    padding-top: 40px;


    .slide-track {
        animation: scroll $animationSpeed linear infinite;
        display: flex;
        height: 200px;
    }

    .slide {
        box-shadow: 0 10px 20px -5px rgba(0, 0, 0, .125);
        background: #ddd;
        width: 150px;
        border-radius: 10px;
        background: url("../avatar_book.png");
        background-repeat: no-repeat;
        background-size: cover;
    }
}