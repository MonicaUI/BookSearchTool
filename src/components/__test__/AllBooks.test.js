import { screen } from '@testing-library/react';
import { shallow } from "enzyme";
import { AllBooks } from '../AllBooks';

test('the Home page', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn("useState");
    useStateSpy.mockImplementation((initialState) => [initialState, setState]);
    const wrapper = shallow(<AllBooks />);
    const countValue = screen.getByText()
    expect(countValue.textContent).toBe("0");

});
