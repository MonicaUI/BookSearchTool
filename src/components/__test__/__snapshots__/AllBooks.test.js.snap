// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home page 1`] = `
<div>
  <div
    className="searchbar"
  >
    <input
      className="searchbar__input"
      id="search-input"
      onChange={[Function]}
      placeholder="Book Title"
      type="text"
    />
    <button
      className="searchbar__button"
      id="search-button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="Home-book"
  >
    <p>
       Welcome to Open Library 
    </p>
    <div
      className="trending"
    />
  </div>
  <div />
</div>
`;
